### PR TITLE
add tags for each violation found

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

**Comment by maxice8** [comment]
@Ikke

**Comment by maxice8** [comment]
@maxwell-k example output.

```
TAG:PATH:LINE:MSG
```

```
[AL1]:community/6tunnel/APKBUILD:16:builddir can be removed as it is the default value
[AL2]:community/6tunnel/APKBUILD:26:|| return 1 is not required as set -e is used
[AL2]:community/6tunnel/APKBUILD:27:|| return 1 is not required as set -e is used
[AL2]:community/6tunnel/APKBUILD:32:|| return 1 is not required as set -e is used
[AL2]:community/6tunnel/APKBUILD:34:|| return 1 is not required as set -e is used
[AL13]:community/6tunnel/APKBUILD:19:cd "$builddir" can be removed in phase 'build'
[AL13]:community/6tunnel/APKBUILD:31:cd "$builddir" can be removed in phase 'package'
```

**Comment by maxice8** [comment]
Updated, the changes to the `NEWS` file should provide an overview of what was done.

**Comment by maxwell-k** [comment]
This looks good! I am following along.

How does this tie in with the severity from #10? Might it be something like:

```
SEVERITY:TAG:PATH:LINE:MSG
```

Maybe this PR lands first, but if this is the general direction, then brilliant.

I want to spend some more time on #2 before I start to make use of this new output format; but I think having it stable and documented will make integrating with other tools straightforward. Thank you!

**Comment by maxice8** [comment]
Ok, pushed a new version with a new system based on debian's Lintian, instead of a single tag we have 2, one for the severity of the violation and one for the certainty that the violation found is not a false positive

File: tests/apkbuild-lint.bats [hunk 1]
```diff
@@ -27,7 +27,7 @@ is_travis() {
 
 	run $cmd $apkbuild
 	[[ $status -eq 1 ]]
-	assert_match "${lines[0]}" "builddir"
+	assert_match "${lines[0]}" "[AL1].*:builddir can be removed as it is the default value"
```
Comment (Ikke) [inline tests/apkbuild-lint.bats]: Brackets needs to be escaped, otherwise it's a character class.

**Comment by maxice8** [comment]
On Tue, May 28, 2019 at 09:48:26PM -0700, Kevin Daudt wrote:
> Ikke commented on this pull request.
> 
> 
> 
> > @@ -27,7 +27,7 @@ is_travis() {
>  
>  	run $cmd $apkbuild
>  	[[ $status -eq 1 ]]
> -	assert_match "${lines[0]}" "builddir"
> +	assert_match "${lines[0]}" "[AL1].*:builddir can be removed as it is the default value"
> 
> Brackets needs to be escaped, otherwise it's a character class.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/maxice8/atools/pull/16#pullrequestreview-243037324

Done, thank you

**Pull request merged**